pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -117,31 +96,34 @@ public AggregateOperator(OpChainExecutionContext context, MultiStageOperator inp
       _aggCallSignatureMap = Collections.emptyMap();
     }
 
-    _hasReturnedAggregateBlock = false;
+    // Initialize the aggregation functions
     _colNameToIndexMap = new HashMap<>();
-
-    // Convert groupSet to ExpressionContext that our aggregation functions understand.
-    List<ExpressionContext> groupByExpr = getGroupSet(groupSet);
-
     List<FunctionContext> functionContexts = getFunctionContexts(aggCalls);
-    AggregationFunction[] aggFunctions = new AggregationFunction[functionContexts.size()];
-
-    for (int i = 0; i < functionContexts.size(); i++) {
+    int numFunctions = functionContexts.size();
+    AggregationFunction<?, ?>[] aggFunctions = new AggregationFunction[numFunctions];
+    for (int i = 0; i < numFunctions; i++) {
       aggFunctions[i] = AggregationFunctionFactory.getAggregationFunction(functionContexts.get(i), true);
     }
 
+    // Process the filter argument indices
+    int[] filterArgIds = new int[numFunctions];
+    int maxFilterArgId = -1;
```

Comment:
why did we need this extra integer? isnt the filterArgIds array null/empty indicate theres no filter?

Reply:
I got confused initially as well, then find it is never empty from RelNode. Added max filter arg id to help quickly identify whether there is filter, and to create the cache